### DISCUSSION TITLE
consoleprinter alternative for standard logging

### DISCUSSION THREAD

**Discussion by floli** [wwkimball/yamlpath#163] [General]
Hello,

documentation mentions it's easy to implement a logger that is API compatible to the consoleprinter and uses standard python logging.

I wonder if yamlpath ships with such a logging alternative? Re-implementing is still tedious but required for using it as a library that uses the standard python logger.

Thanks!

**Comment by wwkimball** [comment]
The reference implementation of the command-line tools all use this logger class:  https://github.com/wwkimball/yamlpath/blob/master/yamlpath/wrappers/consoleprinter.py

Note the public API.  You can subclass that one -- overriding the methods you'd like to control -- or write your own logger from scratch by duplicating the public API of that class.
